src/toil/test/src/regularLogTest.py

```diff
@@ -13,12 +13,32 @@
 # limitations under the License.
 import subprocess
 import sys
+import os
+import mimetypes
 from toil.test import ToilTest
 from toil.test.mesos import helloWorld
 
 
 class RegularLogTest(ToilTest):
 
+    def setUp(self):
+        super(RegularLogTest, self).setUp()
+        self.tempDir = self._createTempDir(purpose='tempDir')
+
+    def _getFiles(self, dir):
+        return [f for f in os.listdir(dir) if os.path.isfile(os.path.join(dir, f))]
+
+    def _assertFileTypeExists(self, dir, extension, encoding=None):
+        # an encoding of None implies no compression
+        onlyFiles = self._getFiles(dir)
+        onlyLogs = [f for f in onlyFiles if f.endswith(extension)]
+        assert onlyLogs
+        for log in onlyLogs:
+            mime = mimetypes.guess_type(log)
```

Comment:
I think this only looks at the file name, which would be pointless.

gzip files always begin with the magic header `'\x1f\x8b'`. IMO, that would be sufficient to verify that they are gzip.